<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>EvolutionConfigForm</class>
 <widget class="QWidget" name="EvolutionConfigForm">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>792</width>
    <height>544</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QGridLayout" name="gridLayout">
   <item row="0" column="0">
    <widget class="QWidget" name="widget" native="true">
     <layout class="QGridLayout" name="gridLayout_3">
      <item row="3" column="6">
       <widget class="QComboBox" name="combo_evolutionmode">
        <property name="currentText">
         <string>brain</string>
        </property>
        <item>
         <property name="text">
          <string>brain</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>full</string>
         </property>
        </item>
       </widget>
      </item>
      <item row="11" column="3">
       <widget class="QDoubleSpinBox" name="dspin_pnoderemove">
        <property name="maximum">
         <double>1.000000000000000</double>
        </property>
        <property name="singleStep">
         <double>0.010000000000000</double>
        </property>
        <property name="value">
         <double>0.100000000000000</double>
        </property>
       </widget>
      </item>
      <item row="0" column="6">
       <widget class="QComboBox" name="combo_selection">
        <item>
         <property name="text">
          <string>deterministic-tournament</string>
         </property>
        </item>
       </widget>
      </item>
      <item row="3" column="5">
       <widget class="QLabel" name="label_40">
        <property name="toolTip">
         <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Specify whether you want to do co-evolution or evolve only the brain&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
        <property name="text">
         <string>Evolution Mode</string>
        </property>
       </widget>
      </item>
      <item row="0" column="0">
       <widget class="QLabel" name="label">
        <property name="toolTip">
         <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;table border=&quot;0&quot; style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px;&quot; cellspacing=&quot;2&quot; cellpadding=&quot;0&quot;&gt;&lt;tr&gt;&lt;td/&gt;&lt;td&gt;&lt;p&gt;Specify the parameters for (μ+/,λ) evolution strategy. mu (μ) = number of parents.&lt;/p&gt;&lt;/td&gt;&lt;/tr&gt;&lt;/table&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
        <property name="text">
         <string>Mu</string>
        </property>
       </widget>
      </item>
      <item row="10" column="3">
       <widget class="QDoubleSpinBox" name="dspin_psubtreeswap">
        <property name="maximum">
         <double>1.000000000000000</double>
        </property>
        <property name="singleStep">
         <double>0.010000000000000</double>
        </property>
        <property name="value">
         <double>0.200000000000000</double>
        </property>
       </widget>
      </item>
      <item row="1" column="6">
       <widget class="QComboBox" name="combo_replacement">
        <item>
         <property name="text">
          <string>plus</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>comma</string>
         </property>
        </item>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="label_3">
        <property name="toolTip">
         <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Number of generations to be evaluated&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
        <property name="text">
         <string>#Generations</string>
        </property>
       </widget>
      </item>
      <item row="4" column="0">
       <widget class="QLabel" name="label_5">
        <property name="text">
         <string>brainSigma</string>
        </property>
       </widget>
      </item>
      <item row="6" column="6">
       <widget class="QSpinBox" name="spin_seed">
        <property name="suffix">
         <string/>
        </property>
        <property name="maximum">
         <number>10000000</number>
        </property>
       </widget>
      </item>
      <item row="7" column="5">
       <widget class="QLabel" name="label_18">
        <property name="text">
         <string>Random Seed</string>
        </property>
       </widget>
      </item>
      <item row="10" column="0">
       <widget class="QLabel" name="label_10">
        <property name="text">
         <string>pSubTreeSwap</string>
        </property>
       </widget>
      </item>
      <item row="6" column="3">
       <widget class="QDoubleSpinBox" name="dspin_pbraincrossover">
        <property name="maximum">
         <double>1.000000000000000</double>
        </property>
        <property name="singleStep">
         <double>0.010000000000000</double>
        </property>
        <property name="value">
         <double>0.000000000000000</double>
        </property>
       </widget>
      </item>
      <item row="0" column="3">
       <widget class="QSpinBox" name="spin_mu">
        <property name="minimum">
         <number>1</number>
        </property>
        <property name="value">
         <number>20</number>
        </property>
       </widget>
      </item>
      <item row="1" column="3">
       <widget class="QSpinBox" name="spin_lambda">
        <property name="minimum">
         <number>1</number>
        </property>
        <property name="value">
         <number>20</number>
        </property>
       </widget>
      </item>
      <item row="8" column="6">
       <widget class="QSpinBox" name="spin_maxbodyparts">
        <property name="suffix">
         <string/>
        </property>
        <property name="maximum">
         <number>1000</number>
        </property>
        <property name="value">
         <number>1000</number>
        </property>
       </widget>
      </item>
      <item row="8" column="5">
       <widget class="QLabel" name="label_69">
        <property name="text">
         <string>maxBodyParts</string>
        </property>
       </widget>
      </item>
      <item row="9" column="6">
       <widget class="QDoubleSpinBox" name="dspin_paddhiddenneuron">
        <property name="maximum">
         <double>1.000000000000000</double>
        </property>
        <property name="singleStep">
         <double>0.010000000000000</double>
        </property>
       </widget>
      </item>
      <item row="10" column="6">
       <widget class="QComboBox" name="combo_evolutionaryalgorithm">
        <property name="currentText">
         <string>Basic</string>
        </property>
        <item>
         <property name="text">
          <string>Basic</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>HyperNEAT</string>
         </property>
        </item>
       </widget>
      </item>
      <item row="9" column="5">
       <widget class="QLabel" name="label_70">
        <property name="text">
         <string>pAddHiddenNeuron</string>
        </property>
       </widget>
      </item>
      <item row="10" column="5">
       <widget class="QLabel" name="label_73">
        <property name="toolTip">
         <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;HyperNEAT can be used only with brain-only evolutions&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
        <property name="text">
         <string>Algorithm</string>
        </property>
       </widget>
      </item>
      <item row="8" column="3">
       <widget class="QDoubleSpinBox" name="dspin_psubtreeremove">
        <property name="maximum">
         <double>1.000000000000000</double>
        </property>
        <property name="singleStep">
         <double>0.010000000000000</double>
        </property>
        <property name="value">
         <double>0.100000000000000</double>
        </property>
       </widget>
      </item>
      <item row="0" column="5">
       <widget class="QLabel" name="label_13">
        <property name="text">
         <string>Selection</string>
        </property>
       </widget>
      </item>
      <item row="7" column="0">
       <widget class="QLabel" name="label_7">
        <property name="text">
         <string>pNodeInsert</string>
        </property>
       </widget>
      </item>
      <item row="8" column="0">
       <widget class="QLabel" name="label_8">
        <property name="text">
         <string>pSubTreeRemove</string>
        </property>
       </widget>
      </item>
      <item row="9" column="0">
       <widget class="QLabel" name="label_9">
        <property name="text">
         <string>pSubTreeDuplicate</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="label_2">
        <property name="toolTip">
         <string>Specify the parameters for (μ+/,λ) evolution strategy. lambda (λ) = number of offspring.</string>
        </property>
        <property name="text">
         <string>Lambda</string>
        </property>
       </widget>
      </item>
      <item row="4" column="3">
       <widget class="QDoubleSpinBox" name="dspin_brainsigma">
        <property name="maximum">
         <double>1.000000000000000</double>
        </property>
        <property name="singleStep">
         <double>0.010000000000000</double>
        </property>
        <property name="value">
         <double>0.700000000000000</double>
        </property>
       </widget>
      </item>
      <item row="11" column="0">
       <widget class="QLabel" name="label_11">
        <property name="text">
         <string>pNodeRemove</string>
        </property>
       </widget>
      </item>
      <item row="6" column="5">
       <widget class="QLabel" name="label_17">
        <property name="text">
         <string>Seed</string>
        </property>
       </widget>
      </item>
      <item row="1" column="5">
       <widget class="QLabel" name="label_27">
        <property name="toolTip">
         <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Pick a replacement strategy&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
        <property name="text">
         <string>Replacement</string>
        </property>
       </widget>
      </item>
      <item row="7" column="6">
       <widget class="QCheckBox" name="check_randomseed">
        <property name="text">
         <string/>
        </property>
       </widget>
      </item>
      <item row="2" column="5">
       <widget class="QLabel" name="label_28">
        <property name="toolTip">
         <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Pick the tournament size for deterministic tournament selection&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
        <property name="text">
         <string>Tournament Size</string>
        </property>
       </widget>
      </item>
      <item row="2" column="6">
       <widget class="QSpinBox" name="spin_tournament">
        <property name="minimum">
         <number>1</number>
        </property>
        <property name="value">
         <number>2</number>
        </property>
       </widget>
      </item>
      <item row="12" column="0">
       <widget class="QLabel" name="label_12">
        <property name="text">
         <string>pParameterModify</string>
        </property>
       </widget>
      </item>
      <item row="0" column="4">
       <spacer name="horizontalSpacer">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item row="3" column="0">
       <widget class="QLabel" name="label_4">
        <property name="text">
         <string>pBrainMutate</string>
        </property>
       </widget>
      </item>
      <item row="6" column="0">
       <widget class="QLabel" name="label_6">
        <property name="text">
         <string>pBrainCrossover</string>
        </property>
       </widget>
      </item>
      <item row="12" column="3">
       <widget class="QDoubleSpinBox" name="dspin_pparemetermodify">
        <property name="maximum">
         <double>1.000000000000000</double>
        </property>
        <property name="singleStep">
         <double>0.010000000000000</double>
        </property>
        <property name="value">
         <double>0.100000000000000</double>
        </property>
       </widget>
      </item>
      <item row="7" column="3">
       <widget class="QDoubleSpinBox" name="dspin_pnodeinsert">
        <property name="maximum">
         <double>1.000000000000000</double>
        </property>
        <property name="singleStep">
         <double>0.010000000000000</double>
        </property>
        <property name="value">
         <double>0.500000000000000</double>
        </property>
       </widget>
      </item>
      <item row="2" column="3">
       <widget class="QSpinBox" name="spin_generations">
        <property name="minimum">
         <number>1</number>
        </property>
        <property name="maximum">
         <number>10000</number>
        </property>
        <property name="value">
         <number>50</number>
        </property>
       </widget>
      </item>
      <item row="3" column="3">
       <widget class="QDoubleSpinBox" name="dspin_pbrainmutate">
        <property name="maximum">
         <double>1.000000000000000</double>
        </property>
        <property name="singleStep">
         <double>0.010000000000000</double>
        </property>
        <property name="value">
         <double>0.300000000000000</double>
        </property>
       </widget>
      </item>
      <item row="9" column="3">
       <widget class="QDoubleSpinBox" name="dspin_psubtreeduplicate">
        <property name="maximum">
         <double>1.000000000000000</double>
        </property>
        <property name="singleStep">
         <double>0.010000000000000</double>
        </property>
        <property name="value">
         <double>0.200000000000000</double>
        </property>
       </widget>
      </item>
      <item row="13" column="3">
       <spacer name="verticalSpacer">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>20</width>
          <height>40</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections/>
</ui>
